From ce85068760a799efa5fbd7dedd602ce74d76f54d Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Fri, 10 Nov 2017 13:59:49 +0800
Subject: [PATCH 2892/5242] MLK-16750-7 arm: dts: add dedicated dts for OP-TEE

commit  173e7f7aca947306614dae3fadd592183e226087 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add dedicated dts for OP-TEE.
The modification is about ocram partition. The default dtsi
use lower ocram area for low power things. But i.MX
chips TZASC could only protect OCRAM from high address to low address.
So modify the OCRAM space for OP-TEE.

The PSCI/OPTEE firmware node will be dynamically created in OP-TEE.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                    |   12 ++++++++++++
 arch/arm/boot/dts/imx6dl-sabreauto-optee.dts  |   13 +++++++++++++
 arch/arm/boot/dts/imx6dl-sabresd-optee.dts    |   13 +++++++++++++
 arch/arm/boot/dts/imx6q-sabreauto-optee.dts   |   14 ++++++++++++++
 arch/arm/boot/dts/imx6q-sabresd-optee.dts     |   14 ++++++++++++++
 arch/arm/boot/dts/imx6qp-sabreauto-optee.dts  |   14 ++++++++++++++
 arch/arm/boot/dts/imx6qp-sabresd-optee.dts    |   14 ++++++++++++++
 arch/arm/boot/dts/imx6sx-sabreauto-optee.dts  |   23 +++++++++++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb-optee.dts        |   23 +++++++++++++++++++++++
 arch/arm/boot/dts/imx6ul-14x14-evk-optee.dts  |   14 ++++++++++++++
 arch/arm/boot/dts/imx6ul-9x9-evk-optee.dts    |   13 +++++++++++++
 arch/arm/boot/dts/imx6ull-14x14-evk-optee.dts |   13 +++++++++++++
 arch/arm/boot/dts/imx7d-sdb-optee.dts         |   22 ++++++++++++++++++++++
 13 files changed, 202 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6dl-sabreauto-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6dl-sabresd-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6q-sabreauto-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6q-sabresd-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6qp-sabreauto-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6qp-sabresd-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6sx-sabreauto-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6sx-sdb-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-14x14-evk-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6ul-9x9-evk-optee.dts
 create mode 100644 arch/arm/boot/dts/imx6ull-14x14-evk-optee.dts
 create mode 100644 arch/arm/boot/dts/imx7d-sdb-optee.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index bbeaf63..634a439 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -413,11 +413,13 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6dl-sabreauto-enetirq.dtb \
 	imx6dl-sabreauto-flexcan1.dtb \
 	imx6dl-sabreauto-gpmi-weim.dtb \
+	imx6dl-sabreauto-optee.dtb \
 	imx6dl-sabrelite.dtb \
 	imx6dl-sabresd.dtb \
 	imx6dl-sabresd-btwifi.dtb \
 	imx6dl-sabresd-hdcp.dtb \
 	imx6dl-sabresd-ldo.dtb \
+	imx6dl-sabresd-optee.dtb \
 	imx6dl-savageboard.dtb \
 	imx6dl-ts4900.dtb \
 	imx6dl-ts7970.dtb \
@@ -510,12 +512,14 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6q-sabreauto-enetirq.dtb \
 	imx6q-sabreauto-flexcan1.dtb \
 	imx6q-sabreauto-gpmi-weim.dtb \
+	imx6q-sabreauto-optee.dtb \
 	imx6q-sabrelite.dtb \
 	imx6q-sabresd.dtb \
 	imx6q-sabresd-btwifi.dtb \
 	imx6q-sabresd-hdcp.dtb \
 	imx6q-sabresd-ldo.dtb \
 	imx6q-sabresd-enetirq.dtb \
+	imx6q-sabresd-optee.dtb \
 	imx6q-savageboard.dtb \
 	imx6q-sbc6x.dtb \
 	imx6q-tbs2910.dtb \
@@ -544,11 +548,13 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6qp-sabreauto-ecspi.dtb \
 	imx6qp-sabreauto-flexcan1.dtb \
 	imx6qp-sabreauto-gpmi-weim.dtb \
+	imx6qp-sabreauto-optee.dtb \
 	imx6qp-sabresd.dtb \
 	imx6qp-sabresd-btwifi.dtb \
 	imx6qp-sabresd-hdcp.dtb \
 	imx6qp-sabresd-ldo.dtb \
 	imx6qp-sabresd-ldo-pcie-cert.dtb \
+	imx6qp-sabresd-optee.dtb \
 	imx6qp-tx6qp-8037.dtb \
 	imx6qp-tx6qp-8037-mb7.dtb \
 	imx6qp-tx6qp-8137.dtb \
@@ -567,9 +573,11 @@ dtb-$(CONFIG_SOC_IMX6SX) += \
 	imx6sx-14x14-arm2.dtb \
 	imx6sx-sabreauto.dtb \
 	imx6sx-sabreauto-m4.dtb \
+	imx6sx-sabreauto-optee.dtb \
 	imx6sx-sdb-reva.dtb \
 	imx6sx-sdb-reva-ldo.dtb \
 	imx6sx-sdb-sai.dtb \
+	imx6sx-sdb-optee.dtb \
 	imx6sx-sdb.dtb \
 	imx6sx-sdb-btwifi.dtb \
 	imx6sx-sdb-emmc.dtb \
@@ -601,6 +609,7 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-14x14-evk-emmc.dtb \
 	imx6ul-14x14-evk-gpmi-weim.dtb \
 	imx6ul-14x14-evk-usb-certi.dtb \
+	imx6ul-14x14-evk-optee.dtb \
 	imx6ul-geam.dtb \
 	imx6ul-isiot-emmc.dtb \
 	imx6ul-isiot-nand.dtb \
@@ -621,6 +630,7 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-14x14-evk-pf1550.dtb \
 	imx6ul-9x9-evk.dtb \
 	imx6ul-9x9-evk-csi.dtb \
+	imx6ul-9x9-evk-optee.dtb \
 	imx6ul-9x9-evk-btwifi.dtb \
 	imx6ul-9x9-evk-btwifi-oob.dtb \
 	imx6ul-9x9-evk-ldo.dtb
@@ -649,6 +659,7 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-14x14-evk-emmc.dtb \
 	imx6ull-14x14-evk-gpmi-weim.dtb \
 	imx6ull-14x14-evk-usb-certi.dtb \
+	imx6ull-14x14-evk-optee.dtb \
 	imx6ull-9x9-evk.dtb \
 	imx6ull-9x9-evk-btwifi.dtb \
 	imx6ull-9x9-evk-btwifi-oob.dtb \
@@ -676,6 +687,7 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sdb-sht11.dtb \
 	imx7d-sdb-qspi.dtb \
 	imx7d-sdb-mipi-dsi.dtb \
+	imx7d-sdb-optee.dtb \
 	imx7d-sdb-reva.dtb \
 	imx7d-sdb-reva-epdc.dtb \
 	imx7d-sdb-reva-gpmi-weim.dtb \
diff --git a/arch/arm/boot/dts/imx6dl-sabreauto-optee.dts b/arch/arm/boot/dts/imx6dl-sabreauto-optee.dts
new file mode 100644
index 0000000..2362784
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-sabreauto-optee.dts
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dl-sabreauto.dts"
+
+&ocram {
+	reg = <0x00905000 0x13000>;
+};
diff --git a/arch/arm/boot/dts/imx6dl-sabresd-optee.dts b/arch/arm/boot/dts/imx6dl-sabresd-optee.dts
new file mode 100644
index 0000000..7b5af9a
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-sabresd-optee.dts
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6dl-sabresd.dts"
+
+&ocram {
+	reg = <0x00905000 0x13000>;
+};
diff --git a/arch/arm/boot/dts/imx6q-sabreauto-optee.dts b/arch/arm/boot/dts/imx6q-sabreauto-optee.dts
new file mode 100644
index 0000000..6b684bd
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sabreauto-optee.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6q-sabreauto.dts"
+
+&ocram {
+	/* Reserve the high 32K for OP-TEE */
+	reg = <0x00905000 0x33000>;
+};
diff --git a/arch/arm/boot/dts/imx6q-sabresd-optee.dts b/arch/arm/boot/dts/imx6q-sabresd-optee.dts
new file mode 100644
index 0000000..1fa1094
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-sabresd-optee.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6q-sabresd.dts"
+
+&ocram {
+	/* Reserve the high 32K for OP-TEE */
+	reg = <0x00905000 0x33000>;
+};
diff --git a/arch/arm/boot/dts/imx6qp-sabreauto-optee.dts b/arch/arm/boot/dts/imx6qp-sabreauto-optee.dts
new file mode 100644
index 0000000..1d9cdcd
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qp-sabreauto-optee.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6qp-sabreauto.dts"
+
+/* TODO: This needs to be changed to use OCRAM3 */
+&ocram {
+	reg = <0x00905000 0x33000>;
+};
diff --git a/arch/arm/boot/dts/imx6qp-sabresd-optee.dts b/arch/arm/boot/dts/imx6qp-sabresd-optee.dts
new file mode 100644
index 0000000..8a036e9
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qp-sabresd-optee.dts
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6qp-sabresd.dts"
+
+/* TODO: This needs to be changed to use OCRAM3 */
+&ocram {
+	reg = <0x00905000 0x33000>;
+};
diff --git a/arch/arm/boot/dts/imx6sx-sabreauto-optee.dts b/arch/arm/boot/dts/imx6sx-sabreauto-optee.dts
new file mode 100644
index 0000000..a84a09e
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-sabreauto-optee.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sabreauto.dts"
+
+&ocram {
+	reg = <0x00905000 0x1b000>;
+};
+
+&ocrams_ddr {
+	reg = <0x00904000 0x1000>;
+};
+
+&ocrams {
+	/* 0x8f8000 are used by optee */
+	reg = <0x00900000 0x4000>;
+	clocks = <&clks IMX6SX_CLK_OCRAM>;
+};
diff --git a/arch/arm/boot/dts/imx6sx-sdb-optee.dts b/arch/arm/boot/dts/imx6sx-sdb-optee.dts
new file mode 100644
index 0000000..8c247cf
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-sdb-optee.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb.dts"
+
+&ocram {
+	reg = <0x00905000 0x1b000>;
+};
+
+&ocrams_ddr {
+	reg = <0x00904000 0x1000>;
+};
+
+&ocrams {
+	/* 0x8f8000 are used by optee */
+	reg = <0x00900000 0x4000>;
+	clocks = <&clks IMX6SX_CLK_OCRAM>;
+};
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk-optee.dts b/arch/arm/boot/dts/imx6ul-14x14-evk-optee.dts
new file mode 100644
index 0000000..a764769
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk-optee.dts
@@ -0,0 +1,14 @@
+
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ul-14x14-evk.dts"
+
+&ocram {
+	reg = <0x00905000 0x13000>;
+};
diff --git a/arch/arm/boot/dts/imx6ul-9x9-evk-optee.dts b/arch/arm/boot/dts/imx6ul-9x9-evk-optee.dts
new file mode 100644
index 0000000..cfde6b1
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-9x9-evk-optee.dts
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ul-9x9-evk.dts"
+
+&ocram {
+	reg = <0x00905000 0x13000>;
+};
diff --git a/arch/arm/boot/dts/imx6ull-14x14-evk-optee.dts b/arch/arm/boot/dts/imx6ull-14x14-evk-optee.dts
new file mode 100644
index 0000000..4027c54
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-14x14-evk-optee.dts
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ull-14x14-evk.dts"
+
+&ocram {
+	reg = <0x00905000 0x13000>;
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb-optee.dts b/arch/arm/boot/dts/imx7d-sdb-optee.dts
new file mode 100644
index 0000000..9302a2c
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-sdb-optee.dts
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+#include "imx7d-sdb.dts"
+
+&ocram {
+	reg = <0x00905000 0x1b000>;
+};
+
+&ocrams_ddr {
+	reg = <0x00904000 0x1000>;
+};
+
+&ocrams {
+	/* 0x180000 are used by optee */
+	reg = <0x00900000 0x4000>;
+	clocks = <&clks IMX7D_OCRAM_CLK>;
+};
-- 
1.7.9.5

